template/src/app.test.ts

```diff
@@ -0,0 +1,16 @@
+import { screen } from "@testing-library/dom";
+import { start } from "tram-one";
+import app from "./app";
+
+describe("app", () => {
+  afterEach(() => {
+    document.body.innerHTML = "";
+  });
+
+  it("should render a link to the tram-one site", () => {
+    start(app, "body");
+
+    const link = screen.queryByText("Learn Tram-One");
+    expect(link).toHaveAttribute("href", "https://tram-one.io");
```

Comment:
changed this test to just start Tram-One with the app as the component. This ensures that all the scaffolding that is required is actually laid out, whereas the previous test code would have missed that.

Reply:
additionally (and the primary purpose for changing this), calling app on it's own actually caused a type error, first that the component needed two parameters (props and children) and then that it wasn't of type HTMLElement, requiring the following code to suffice
```typescript
const renderedApp = app(undefined, undefined) as unknown as HTMLElement;
```

While thinking about this, I realized that `start` was better for all the other things that could break here, so I think this solution is fine. 

I do need to go back to Tram-One and make those parameters optional and see what we can do to make `TramOneElement` better map to `HTMLElement`.